Tasks/Gradle/sonarqube-common/sonarqube-common.ts

```diff
@@ -180,22 +182,38 @@ function isNullOrEmpty(str) {
 }
 
 // Creates the string that comprises the build summary text, given the location of the /sonar build folder.
-function createSonarQubeBuildSummary(sqBuildFolder: string): string {
-    // Task report is not created for PR builds
+function createSonarQubeBuildSummary(sqBuildFolder:string):Q.Promise<string> {
+    // Task report is not created for PR builds - inform the user in the build summary
     if (isPrBuild()) {
+        tl.debug('Pull request build, will not attempt to fetch detailed SonarQube report');
         // Looks like: Detailed SonarQube reports are not available for pull request builds.
-        return tl.loc('sqAnalysis_BuildSummaryNotAvailableInPrBuild');
+        return Q.when(tl.loc('sqAnalysis_BuildSummaryNotAvailableInPrBuild'));
     }
 
-    var taskReport: TaskReport = getSonarQubeTaskReport(sqBuildFolder);
-    if (!taskReport) {
-        throw new Error(tl.loc('sqAnalysis_TaskReportInvalid'));
-    }
+    // Wait for analysis to finish so that we can get analysis information from the SonarQube server
+    var timeout:number = 300;
+    var delay:number = 1;
 
-    var linkToDashBoard: string = createLinkToSonarQubeDashboard(sqBuildFolder);
+    var sqRunSettings:SonarQubeRunSettings = getSonarQubeTaskReport(sqBuildFolder);
 
-    // Put the quality gate status and dashboard link sections together with the Markdown newline
-    return linkToDashBoard;
+    var sqServer:ISonarQubeServer = new SonarQubeServer();
+    var analysisMetrics:SonarQubeMetrics = new SonarQubeMetrics(sqServer, sqRunSettings.ceTaskId, timeout, delay);
+    var sqReportBuilder:SonarQubeReportBuilder = new SonarQubeReportBuilder(sqRunSettings, analysisMetrics);
+
+    // Looks like: "[Detailed SonarQube report >](https://mySQserver:9000/dashboard/index/foo "foo Dashboard")"
+    var linkToDashBoard:string = sqReportBuilder.createLinkToSonarQubeDashboard();
```

Comment:
The sections of the report and the binding of these sections should also be part of the ReportBuilder and not exposed here.